nixos/modules/services/development/lorri.nix

```diff
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.lorri;
+  socketUnit = "lorri";
+  socketPath = "lorri/daemon.socket";
+in {
+  options = {
+    services.lorri = {
+      enable = lib.mkOption {
+        default = false;
+        type = lib.types.bool;
+        description = ''
+          This option enables a systemd socket unit that listens on the
+          well-known address used by lorri, a nix-shell replacement for project
+          development. When a lorri client connects to the socket, systemd
+          starts the lorri daemon as a service.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.user.sockets.${socketUnit} = {
+      description = "Socket for Lorri Daemon";
+      wantedBy = [ "sockets.target" ];
+      unitConfig = {
+        ConditionUser = "!@system";
+      };
+      socketConfig = {
+        ListenStream = "%t/${socketPath}";
+        RuntimeDirectory = "lorri";
```

Comment:
Shouldn't that be part of the `lorri.service` unit? I'm not sure if it works here at all…

Reply:
The directory of the _socket_ is created using https://docs.rs/directories/2.0.2/directories/struct.ProjectDirs.html#method.runtime_dir, hence the socket unit's `RuntimeDirectory` must be set in this way.

Reply:
I assume we should set `RuntimeDirectory` in `lorri.service` too - it also requires `%t/lorri` - just in case people manually start `lorri.service`, without `lorri.socket` being enabled.

Reply:
Done, thanks!